unit-10/script.js

```diff
@@ -0,0 +1,188 @@
+// lesson 1:
+// In một bảng nhân của một số bất kỳ với các số từ 1 tới 10 và hiển thị kết quả
+
+function getMultiplyTable(number) {
+  let result =''
+  if (typeof(number) !== 'number') {
+    result = 'Dữ liệu nhập không hợp lệ: Phải là một số'
+  } else {
+    result = 'In bang nhan:\n' 
+    for (let i = 1; i <= 10; i++) {
+      result += '   ' + number + ' x ' + i + ' = ' + number * i + '\n'
+    }
+  }
+  return result
+}
+
+// lesson 2:
+// Hàm nhận tham số là n (số nguyên dương), 1 <= n <= 30 In các số chẵn từ 1 đến n Vd n = 6: in ra: 2 4 6
+
+function getEvenNumber(number) {
+  let result = ''
+  if (!(Number.isInteger(number)) || number < 1 || number > 30) {
+    result = 'Dữ liệu nhập không hợp lệ: Phải là một số nguyên dương 1 <= n <= 30'
+  } else {
+    for (let i = 2; i <= number; i +=2) {
+      result += i + ' '
+    }
+  }
+  return result
+}
+
+// lesson 3:
+// Hàm nhận tham số là n (số nguyên dương), 1 <= n <= 30 Tính tổng từ 1 đến n
+// vd nhập vào 6: Tổng = 1 + 2 + 3 + 4 + 5 + 6 = 21
+
+function getSumNumber(number) {
+  let result = ''
+  if (!(Number.isInteger(number)) || number < 1 || number > 30) {
+    result = 'Dữ liệu nhập không hợp lệ: Phải là một số nguyên dương 1 <= n <= 30'
+  } else {
+    let sum = 0
+
+    for (let i = 1; i <= number; i++) {
+      sum += i
+      if (i !== number) {
+        result += i + ' + '
+      } else {
+        result += i
+      }
+    }
+    result = 'Tổng = ' + result + ' = ' + sum
+  }
+  return result
+}
+
+// lesson 4:
+// Hàm nhận tham số là n (số nguyên dương), 1 <= n <= 30 In ra n! (giai thừa của n)
+
+function getFactorial(number) {
+  let result = ''
+  if (!(Number.isInteger(number)) || number < 1 || number > 30) {
+    result = 'Dữ liệu nhập không hợp lệ: Phải là một số nguyên dương 1 <= n <= 30'
+  } else {
+    factorial = 1
+    for (let i = 1; i <= number; i++) {
+      factorial *= i
+    }
+    result = number + '! = ' + factorial
+  }
+  return result
+}
```

Comment:
```suggestion
const validate = number => Number.isInteger(number) && number > 0 && number < 31
function getFactorial (number) {
  if (!validate(number)) {
    return 'Dữ liệu nhập không hợp lệ: Phải là một số nguyên dương 1 <= n <= 30'
  }
  
  let factorial = 1
  for (let i = 1; i <= number; i++) {
    factorial *= i
  }

  return factorial
}
```

Viết hàm `validate` để có thể dùng ở nhiều chỗ, hạn chế viết `if else` lại